### ISSUE TITLE
fix(diagnostics): multilayer QG stretching term is missing a factor f0 (dimensionally inconsistent; 10⁴× error for realistic f0)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#282]
Part of #275.

## Summary

`potential_vorticity_multilayer` (`finitevolx/_src/operators/diagnostics.py:765-810`) and `QGPotentialVorticity2D.multilayer` (`diagnostic_operators.py`) compute

```
q_k = ∇²ψ_k / f0 + β(y − y0) / f0 − (A · ψ)_k
```

where `A` is documented as the output of `build_coupling_matrix(H, g')`, i.e. `A = 1/(H g')` with units s²/m². The first two terms are dimensionless; `A·ψ` has units of seconds. The reference implementation (louity/MQGeometry `qgm.py`) uses `q = ∇²ψ − f0²·A·ψ + β(y − y0)`; dividing by f0 to match finitevolX's normalisation gives `∇²ψ/f0 − f0·A·ψ + β(y−y0)/f0`. The code omits the `f0`.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp, numpy as np
from finitevolx import build_coupling_matrix, potential_vorticity_multilayer, CartesianGrid2D, Difference2D
H = jnp.array([400.0, 1100.0]); gp = jnp.array([9.81, 0.025]); f0 = 9.375e-5; beta = 1.75e-11
A = build_coupling_matrix(H, gp)
N = 66; L = 1.0e6; g = CartesianGrid2D.from_interior(N-2, N-2, Lx=L, Ly=L)
x = (jnp.arange(N) - 0.5) * g.dx; y = (jnp.arange(N) - 0.5) * g.dy
psi = 1e4 * jnp.stack([jnp.sin(2*jnp.pi*x[None,:]/L) * jnp.sin(2*jnp.pi*y[:,None]/L)] * 2)
lap = jnp.stack([Difference2D(g).laplacian(p) for p in psi])
print("max|lap psi / f0|     :", float(jnp.abs(lap / f0).max()))
print("max|A psi| (as coded) :", float(jnp.abs(jnp.einsum('kl,lji->kji', A, psi)).max()))
print("max|f0 A psi| (MQG/f0):", float(jnp.abs(f0 * jnp.einsum('kl,lji->kji', A, psi)).max()))
```

```
max|lap psi / f0|     : 8.40e-03
max|A psi| (as coded) : 5.03e+02
max|f0 A psi| (MQG/f0): 4.71e-02
ratio coded/expected = 1/f0 = 10666.7
max|q_fvx - q_mqg/f0| / max|q_mqg/f0| = 3.44e+03
```

The only test, `tests/test_diagnostics.py::test_multilayer_stretching_term`, uses `f0 = 1.0`, which hides the missing factor. `tests/test_scale_covariance.py` does not cover this function.

## Root cause

`diagnostics.py:806`: `return qg_pv_layer(psi) - stretching_term(A, psi)`.

## Proposed fix

`return qg_pv_layer(psi) - f0 * stretching_term(A, psi)`, and state in the docstring that `A` is the raw `build_coupling_matrix` output. (Alternative: document that callers must pass `f0·A`; the first is less error-prone.) Same change in `QGPotentialVorticity2D.multilayer`.

## Acceptance criteria

- [ ] Test with physical `f0 = 9.375e-5`, `H = [400, 1100]`, `g' = [9.81, 0.025]` against the MQGeometry formula divided by f0: relative error < 1e-12.
- [ ] Scale-covariance test: rescaling `f0` by 10 changes the stretching term by 10.
- [ ] Docstring formula corrected in both places; `docs/notebooks/qg_1p5_layer.py` re-checked (it builds the 1.5-layer term by hand).
